cmd/livepeer/livepeer.go

```diff
@@ -284,6 +284,7 @@ func main() {
 		return
 	}
 
+	watcherErr := make(chan error)
```

Comment:
I think I understand, but just to make sure: 

Considering the context that we currently have two watchers that potentially return an error when watch is invoked (but not during the for loop itself) , if both error at around the same time we're not correctly terminating whichever of the two goroutines passed an error into the channel last? 

Does that mean that we could resolve this by simply having a seperate channel for the blockwatcher as well, without necessarily wrapping it in a seperate goRoutine that sends the first error encountered into a more general channel? 

Would another approach also work, such as setting a buffer of `2` considering we have two watchers that could return an error into this channel? I know this wouldn't be the most elegant approach but would it work? I think so because the value is sent into the buffer, but the goroutine doesn't care whether it is handled since because there is room in the buffer it doesn't block? 

To me the additional goroutine seemed like a useful approach if we were to use a helper, a little less so if we keep all the logic inside of `main()`, is that an incorrect statement to make? 

That being said I'll introduce the changes as per the suggestion you made. Above questions are more for the purpose of personal  clarification and insight.

Reply:
I think this should do the trick then ? 0929eee